### PR TITLE
Fix mill testing on mill 0.11 and add mill to README and comments

### PR DESCRIPTION
**Pull request by jiegec**
mill 0.11 has changed the way to write test module, see https://github.com/com-lihaoyi/mill/releases/tag/0.11.0

### CODE REVIEW

**Review by jackkoenig — APPROVED** [review]
LGTM Thanks! Sorry I missed the email notification about this PR.

File: .github/workflows/template-cleanup.yml [hunk 1]
```diff
@@ -30,7 +30,7 @@ jobs:
           GROUP="com.github.$SAFE_ORG"
 
           # Replace placeholders
-          sed -i "s/%NAME%/$NAME/g" build.sbt build.sc README.md
+          sed -i "s/%NAME%/$NAME/g" build.sbt build.sc README.md src/test/scala/gcd
```
Comment (towoe) [inline .github/workflows/template-cleanup.yml:33]: @jackkoenig template creation fails for me with "sed: couldn't edit src/test/scala/gcd: not a regular file".
Maybe use 'src/test/scala/gcd/*'?

Comment (jiegec) [reply .github/workflows/template-cleanup.yml:33]: Oops, my fault, you are right.

Comment (jackkoenig) [reply .github/workflows/template-cleanup.yml:33]: Good catch, thanks @towoe! I'll push a fix.

**Pull request merged**